guides/keras_nlp/transformer_pretraining.py

```diff
@@ -0,0 +1,481 @@
+our ability to generalize to unseen examples.
+
+Enter **pre-training**, which will allow us to learn on a larger corpus, and transfer our
+knowledge to the `SST-2` task. And enter **KerasNLP**, which will allow us to pre-train a
+particularly powerful model, the transformer, with ease.
+"""
+
+"""
+## Pre-training
+
+To beat our baseline, we need to leverage the `WikiText103` dataset, an unlabeled
+collection of wikipedia articles that is much bigger than than `SST-2`.
+
+We are going to train a *transformer* model, a highly expressive model which will learn
+to embed each word in our input as a low dimentional vector. Our wikipedia dataset has no
+labels, so we will use an unsupervised training objective called the *Masked Language
+Modeling* (MLM) ojective.
+
+Essentially, we will be playing a big game of "guess the missing word." For each input
+sample we will obscure 20% of our input data, and train our model to predict the parts we
+covered up.
+
+Note that we aren't following any particular transformer architecture here, but you can
+think of our approach as a simplified
+[BERT](https://en.wikipedia.org/wiki/BERT_(language_model)) model.
+"""
+
+"""
+### Preprocess data for the MLM task
+
+Our text preprocessing for the MLM task will occur in two stages.
+ - First, we tokenize input text into integer sequences of token ids.
+ - Second, we mask certain positions in our input to predict on.
+
+To tokenize, we can use a `keras_nlp.tokenizer.Tokenizer`--the KerasNLP building block
+for transforming text into sequences of integer token ids.
+
+In particular, we will use `keras_nlp.tokenizers.WordPieceTokenizer` which does
+*sub-word* tokenization. Sub-word tokenization is popular when training models on large
+text corpora. It offers a good trade off--you don't need a massive vocabulary to encode
+every word in your corpus, but you can still encode common words in their entirety.
+
+The second thing we need to do is mask our input for the MLM task. To do this, we can use
+`keras_nlp.layers.MLMMaskGenerator`, which will randomly select a set of tokens in each
+input and mask them out.
+
+The tokenizer and the masking layer can both be used inside a call to
+[tf.data.Dataset.map](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#map).
+This will allow us to efficiently pre-compute each batch on the CPU, while our GPU or TPU
+works on training with the batch that came before. And because our masking layer will
+choose new words to mask each time, each epoch over our dataset will give us a totally
+new set of labels to train on.
+"""
+
+# Setting sequence_length will trim or pad the token outputs to shape
+# (batch_size, SEQ_LENGTH).
+tokenizer = keras_nlp.tokenizers.WordPieceTokenizer(
+    vocabulary="bert_vocab_uncased.txt", sequence_length=SEQ_LENGTH,
+)
+# Setting mask_selection_length will trim or pad the mask outputs to shape
+# (batch_size, PREDICTIONS_PER_SEQ).
+masker = keras_nlp.layers.MLMMaskGenerator(
```

Comment:
We print the sorted tuple of output later (and it's a pretty bulky print). I don't I think we should print it twice, it would take up too much vertical space.